### PR TITLE
show multiple docstrings in REPL more clearly

### PR DESCRIPTION
**Pull request by KristofferC**
This is done by printing a vertical line next to each separate docstring

Before (hard to see where one docstring ends and the other starts):

screen shot 2018-01-20 at 12 05 38

After:

screen shot 2018-01-20 at 12 05 13

This line is only showed when there are multiple docstrings, otherwise print as usual.

### CODE REVIEW

**Comment by JeffBezanson** [comment]
Maybe a horizontal line between them would make more sense?

**Comment by KristofferC** [comment]
Alright, I change to that. But then I am not changing again! ;;)

**Comment by nalimilan** [comment]
I think we had suggested adding just a very short (2-3 chars) horizontal line in another issue. Have you tried that?

**Comment by KristofferC** [comment]
I tried it but didn't really like it. Too short and it is not visible enough and when you increase the length it looks odd that it just stops in the middle. I think the current version is pretty ok.

**Comment by StefanKarpinski** [comment]
How about bracketed lines like our logging messages?

**Comment by JeffBezanson** [comment]
I think a horizontal rule is the natural thing here. We probably wouldn't add a bracket or vertical line for a single doc string, and it's a bit odd for them to suddenly appear when there are multiple doc strings. With log messages there is usually other surrounding output to distinguish them from.

**Comment by KristofferC** [comment]
Yes, I have tried a lot of versions and I like this the most. Having a long vertical line appear just because there are multiple docstrings is a bit weird. Copy pasting docstrings will also be awkward with vertical lines.

File: test/docs.jl [hunk 1]
```diff
@@ -990,8 +990,8 @@ dynamic_test.x = "test 2"
 @test @doc(dynamic_test) == "test 2 Union{}"
 @test @doc(dynamic_test(::String)) == "test 2 Tuple{String}"
 
-@test Docs._repl(:(dynamic_test(1.0))) == Expr(:escape, Expr(:macrocall, Symbol("@doc"), LineNumberNode(196, doc_util_path), :(dynamic_test(::typeof(1.0)))))
-@test Docs._repl(:(dynamic_test(::String))) == Expr(:escape, Expr(:macrocall, Symbol("@doc"), LineNumberNode(196, doc_util_path), :(dynamic_test(::String))))
+@test Docs._repl(:(dynamic_test(1.0))) == Expr(:escape, Expr(:macrocall, Symbol("@doc"), LineNumberNode(214, doc_util_path), :(dynamic_test(::typeof(1.0)))))
```
Comment (KristofferC) [inline test/docs.jl]: I really dislike these tests...

**Comment by rfourquet** [comment]
x-ref #22870 where this has been also discussed

**Pull request merged**